rpcs3/Emu/Cell/lv2/sys_fs.cpp

```diff
@@ -2064,27 +2064,18 @@ error_code sys_fs_fcntl(ppu_thread& ppu, u32 fd, u32 op, vm::ptr<void> _arg, u32
 	{
 		const auto arg = vm::static_ptr_cast<lv2_file_c0000007>(_arg);
 
-		std::string_view device{arg->device.get_ptr(), arg->device_size};
-
-		// Trim trailing '\0'
-		if (const auto trim_pos = device.find('\0'); trim_pos != umax)
-			device.remove_suffix(device.size() - trim_pos);
-
-		if (device != "CELL_FS_IOS:ATA_HDD"sv)
-		{
-			arg->out_code = CELL_ENOTSUP;
-			return {CELL_ENOTSUP, device};
-		}
-
-		const auto model = g_cfg.sys.hdd_model.to_string();
-		const auto serial = g_cfg.sys.hdd_serial.to_string();
+		arg->out_code = CELL_OK;
 
-		strcpy_trunc(std::span(arg->model.get_ptr(), arg->model_size), model);
-		strcpy_trunc(std::span(arg->serial.get_ptr(), arg->serial_size), serial);
+		if (arg->model_size > 0)
```

Comment:
Try to read once from ps3 memory instead of multiple times from the same address, assume other threads can change the value in between in theory.

Reply:
Fixed.